core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -324,15 +329,20 @@ private void initMetrics() {
                 .build();
     }
 
-    private void initStringSize(Session session) {
-        String configMaxStringSize = configurationService.load("org.hawkular.metrics", "string-size").toBlocking()
+    private void initConfiguration(Session session) {
+        Configuration configuration = configurationService.load("org.hawkular.metrics").toBlocking()
                 .lastOrDefault(null);
+        String configMaxStringSize = configuration.get("string-size");
         if (configMaxStringSize == null) {
             maxStringSize = -1;  // no size limit
         } else {
             maxStringSize = Integer.parseInt(configMaxStringSize);
         }
         log.infoMaxSizeStringMetrics(this.maxStringSize);
+
+        insertRetryMaxDelay = Long.parseLong(configuration.get("ingestion.retry.max-delay", "30000"));
```

Comment:
The value from the cmd line is stored in the database. MetricsServiceImpl then fetches it from the database. We also set defaults in the database so you don't have to pass values via the command line, but command line values will always take precedence.